### PR TITLE
Improve exercise widgets

### PR DESCRIPTION
**Pull request by iHiD**

### CODE REVIEW

File: app/javascript/components/common/ExerciseWidget.tsx [hunk 1]
```diff
@@ -72,50 +76,91 @@ const ReferenceElement = forwardRef<
...
+      />
     )
+    if (solution || exercise.isUnlocked) {
+      const classNames = [
+        'c-exercise-widget',
+        `--${solution ? solution.status : 'available'}`,
+        exercise.isRecommended ? '--recommended' : '',
+        `--${renderAsLink ? 'interactive' : 'static'}`,
+        isSkinny ? '--skinny' : '',
+      ]
```
Comment (ErikSchierboom) [inline app/javascript/components/common/ExerciseWidget.tsx:108]: You could consider indenting by two spaces, to have the closing of the array and the `.filter` be at the same indentation, helping reinforce that they belong together.

```suggestion
          'c-exercise-widget',
          `--${solution ? solution.status : 'available'}`,
          exercise.isRecommended ? '--recommended' : '',
          `--${renderAsLink ? 'interactive' : 'static'}`,
          isSkinny ? '--skinny' : '',
        ]
```

File: app/javascript/components/common/ExerciseWidget.tsx [hunk 2]
```diff
@@ -72,50 +76,91 @@ const ReferenceElement = forwardRef<
...
+          <ExerciseIcon iconUrl={exercise.iconUrl} title={exercise.title} />
+        </div>
+      )
+    } else {
+      const classNames = [
+        'c-exercise-widget',
+        '--locked',
+        `--${renderAsLink ? 'interactive' : 'static'}`,
+        isSkinny ? '--skinny' : '',
+      ]
```
Comment (ErikSchierboom) [inline app/javascript/components/common/ExerciseWidget.tsx:144]:
```suggestion
          'c-exercise-widget',
          '--locked',
          `--${renderAsLink ? 'interactive' : 'static'}`,
          isSkinny ? '--skinny' : '',
        ]
```

**Review by ErikSchierboom — APPROVED** [review]

File: app/javascript/components/common/ExerciseWidget.tsx (hunk 1 above)
Comment (iHiD) [reply app/javascript/components/common/ExerciseWidget.tsx:108]: I'll check but I feel like prettier (or whatever it is) undoes that.

Comment (ErikSchierboom) [reply app/javascript/components/common/ExerciseWidget.tsx:108]: Aha! Weirdness

Comment (iHiD) [reply app/javascript/components/common/ExerciseWidget.tsx:108]: Yeah, I've tried. It undoes it 🤷

**Pull request merged**